test/controllers/HowManyFilesUploadControllerSpec.scala

```diff
@@ -85,11 +89,14 @@ class HowManyFilesUploadControllerSpec extends ControllerSpecBase with DomAssert
   override protected def beforeEach(): Unit = {
     super.beforeEach()
 
+    reset(mockCustomsDeclarationsService, mockUpscanConnector, page)
+
     when(mockCustomsDeclarationsService.batchFileUpload(any(), any(), any())(any())).thenReturn(Future.successful(FileUploadResponse(List())))
+    when(page(any[Form[FileUploadCount]])(any(), any())).thenReturn(HtmlFormat.empty)
   }
 
   override protected def afterEach(): Unit = {
-    reset(mockCustomsDeclarationsService, mockUpscanConnector)
+    reset(mockCustomsDeclarationsService, mockUpscanConnector, page)
```

Comment:
NIT: Why bother resetting both before and after each test (seems wasteful and unnecessary)?

Reply:
Same question again 😂 I prefer to have a guarantee that each test is executed in a 'clean' environment. 
It may seem unnecessary, but I've had some bad experiences because of the lack of it. It is not that wasteful, since mock reset is really quick.